pkg/validate/validate_test.go

```diff
@@ -184,10 +184,10 @@ func TestNoProxyDomainName(t *testing.T) {
 		{"1", true},
 		{"0.0", true},
 		{"1.2.3.4", true},
-		{"1.2.3.4.", false},
-		{"abc.", false},
+		{"1.2.3.4.", true},
```

Comment:
Is everything a string or do IPs have special handling?

Reply:
`1.2.3.4.` is a valid hostname if we allow a trailing dot. [rfc 1123](https://tools.ietf.org/html/rfc1123) is being used as the basis for domain name validation:

> One aspect of host name syntax is hereby changed: the restriction on the first character is relaxed to allow either a letter or a digit.  Host software MUST support this more liberal syntax.

Previously the proceeding dot was stripped prior to rfc 1123 validation, so `.1.2.3.4` was and is still considered a valid name. Now the trailing dot will be stripped prior to rfc 1123 validation, so `.1.2.3.4`, `1.2.3.4.` and `.1.2.3.4.` are also considered valid.

Reply:
The values are also run against `net.ParseCIDR`, but since `.1.2.3.4`, `1.2.3.4.` and `.1.2.3.4.` are considered valid by rfc 1123 after stripping proceeding/trailing dots, they are considered valid.